cmd/local.go

```diff
@@ -0,0 +1,367 @@
+	}
+
+	if err != nil {
+		channels.Failed <- 1
+		channels.Errors <- turnstile.ConcurrentError{
+			RunIdentifier: l.Nonmem.Model,
+			Error:         err,
+			Notes:         fmt.Sprintf("There appears to have been an issue trying to copy %s to %s", l.Nonmem.Model, l.Nonmem.OutputDir),
+		}
+		return
+	}
+
+	//Create Execution Script
+	scriptContents, err := generateScript(nonMemExecutionTemplate, l.Nonmem)
+
+	if err != nil {
+		channels.Failed <- 1
+		channels.Errors <- turnstile.ConcurrentError{
+			RunIdentifier: l.Nonmem.Model,
+			Error:         err,
+			Notes:         "An error occurred during the creation of the executable script for this model",
+		}
+	}
+
+	//rwxr-x---
+	afero.WriteFile(fs, path.Join(l.Nonmem.OutputDir, l.Nonmem.FileName+".sh"), scriptContents, 0750)
+}
+
+//Work describes the Turnstile execution phase -> IE What heavy lifting should be done
+func (l LocalModel) Work(channels *turnstile.ChannelMap) {
+	cerr := executeNonMemJob(executeLocalJob, l.Nonmem)
+
+	if cerr.Error != nil {
+		recordConcurrentError(l.Nonmem.Model, cerr.Notes, cerr.Error, channels)
+	}
+
+}
+
+//Monitor is unimplemented here. It's the 3rd phase of Turnstile execution
+func (l LocalModel) Monitor(channels *turnstile.ChannelMap) {
+	//Do nothing for this implementation
+}
+
+//Cleanup is the last phase of execution, in which computation / hard work is done and we're cleaning up leftover files, copying results around et all.
+func (l LocalModel) Cleanup(channels *turnstile.ChannelMap) {
+	time.Sleep(10 * time.Millisecond)
+	log.Printf("Beginning cleanup phase for model %s\n", l.Nonmem.FileName)
+	fs := afero.NewOsFs()
+
+	//Magical instructions
+	//TODO: Implement flags for mandatory copy and cleanup exclusions
+	pwi := newPostWorkInstruction(l.Nonmem, []string{}, []string{})
+
+	//Copy Up first so that we don't try to move something we remove :)
+	var copied []runner.TargetedFile
+	//log.Printf("Beginning copy-up operations for model %s\n", l.FileName)
+	for _, v := range pwi.FilesToCopy.FilesToCopy {
+
+		source, err := utils.ReadLines(path.Join(pwi.FilesToCopy.CopyFrom, v.File))
+
+		if err != nil {
+			//Just continue. There are potentially files which will not exist based on the values in the list.
+			continue
+		}
+
+		//Let's avoid stuttering from extension extrapolation
+		file, _ := utils.FileAndExt(v.File)
+
+		if file == l.Nonmem.FileName {
+			file = v.File
+		} else {
+			file = l.Nonmem.FileName + "." + v.File
+		}
+
+		err = utils.WriteLines(source, path.Join(pwi.FilesToCopy.CopyTo, file))
+
+		if err != nil {
+			log.Printf("An erorr occurred while attempting to copy the files: File is %s", v.File)
+			continue
+		}
+
+		v.File = l.Nonmem.OutputDir + "/" + v.File
+
+		copied = append(copied, v)
+	}
+
+	//Write to File in original path indicating what all was copied
+	copiedJSON, _ := json.MarshalIndent(copied, "", "    ")
+
+	afero.WriteFile(fs, path.Join(l.Nonmem.OriginalPath, "copied.json"), copiedJSON, 0750)
```

Comment:
Thinking about this, we should only write this out if files are actually copied I think. This will keep from cluttering the root dir with a bunch of essentially empty copies.

Also, more importantly this looks to be incorrect. If you have a file named copied.json, multiple models will keep overwriting that file. It should be `{{.Name}}_copied.json`

Reply:
Both make total sense. We already have the model filename as part of the NonMemModel so we can just concatenate it to the filename. We'll also keep track of copied files, and if any are copied, create the file.